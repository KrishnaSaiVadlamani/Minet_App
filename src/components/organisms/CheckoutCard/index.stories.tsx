import { ComponentStory, ComponentMeta } from '@storybook/react'
import React from 'react'
import CheckoutCard from './index'

export default {
  title: 'Organisms/CheckoutCard',
  component: CheckoutCard,
} as ComponentMeta<typeof CheckoutCard>

const Template: ComponentStory<typeof CheckoutCard> = (args) => (
  <CheckoutCard {...args} />
)



export const BuyCheckoutCard = Template.bind({})

BuyCheckoutCard.args = {
    type: 'buy',
    btcValue: 0.0234510
}