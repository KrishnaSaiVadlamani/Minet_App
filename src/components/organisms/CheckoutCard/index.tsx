import { styled, Box, Grid, Stack } from '@mui/material'
import React from 'react'
import TypographyComponent from '../../atoms/Typography'
import { dollarValue, formatCurrency } from '../../../utils/constants'
import theme from '../../../theme'
import TransactionStepper from '../../molecules/transactionStepper'
import IconComponent from '../../atoms/Icons'
import dottedLine from '../../../assets/icons/orderSummary.svg'
import ButtonComponent from '../../atoms/Button'

interface CheckoutCardProps {
  type: 'buy' | 'sell'
  btcValue: number
  price?: number
  onClick: (arg: any) => void
  amount?: number
}
const StyledBox = styled(Box)({
  width: '527px',
  border: `1px solid ${theme.palette.greyColors.grey100}`,
  borderRadius: '4px',
  background: theme.palette.structural.main,
})

const TypographyGrid = styled(Grid)({
  paddingTop: '28px',
  paddingBottom: '32px',
  alignItems: 'center',
  justifyContent: 'center',
  margin: '0px',
  width: '100%',
  borderBottom: `1px solid ${theme.palette.greyColors.grey100}`,
})

const StepperBox = styled(Box)({
  padding: '24px 0 24px 24px',
  borderBottom: `1px solid ${theme.palette.greyColors.grey100}`,
})

const TotalBox = styled(Stack)({
  padding: '24px 0 70px 24px',
})

const renderOrderSummary = (
  leftValue: string,
  rightValue: string,
  width: string,
  bold: boolean
) => {
  return (
    <Stack direction="row" spacing={0}>
      <TypographyComponent
        variant={'overline'}
        children={leftValue}
        color={theme.palette.textColor.highEmphasis}
      />
      <IconComponent
        src={dottedLine}
        width={width}
        height={'5px'}
        padding={bold ? '0 0 1px 0' : '0 0 4px 0'}
      />
      <TypographyComponent
        variant={bold ? 'body1' : 'overline'}
        children={`${rightValue}`}
        color={theme.palette.textColor.highEmphasis}
        display='flex'
        justifyContent={'end'}
      />
    </Stack>
  )
}

const StyledButtonComponent = styled(ButtonComponent)({
  width: '479px',
  height: '42px',
})

const typographyStyle = {
  color: theme.palette.textColor.mediumEmphasis,
}

const CheckoutCard: React.FC<CheckoutCardProps> = ({
  type,
  btcValue,
  price,
  onClick,
  amount,
}) => {
  price = price ? price : dollarValue
  amount = amount ? amount : btcValue * price
  let total = type === 'buy' ? amount + 1000 : amount - 1000
  return (
    <StyledBox>
      <TypographyGrid container spacing="12px" direction={'column'}>
        <Grid item xs={12}>
          <TypographyComponent
            variant="caption1"
            children={`You are ${type}ing`}
            style={typographyStyle}
          />
        </Grid>
        <Grid item xs={12}>
          <TypographyComponent variant="h6" children={`${btcValue} BTC`} />
        </Grid>
        <Grid item xs={12}>
          <TypographyComponent
            variant="caption1"
            children={`1BTC = ${formatCurrency.format(price)}`}
            style={typographyStyle}
          />
        </Grid>
      </TypographyGrid>

      <StepperBox>
        <TransactionStepper />
      </StepperBox>

      <TotalBox spacing={2} direction={'column'}>
        {renderOrderSummary(
          `${btcValue} BTC`,
          formatCurrency.format(amount),
          '330px',
          false
        )}
        {renderOrderSummary('transaction fee', '$1,000.00', '333px', false)}
        {renderOrderSummary(
          'Total',
          formatCurrency.format(total),
          '350px',
          true
        )}
        <StyledButtonComponent
          variant="contained"
          children={type === 'buy' ? 'BUY NOW' : 'SELL NOW'}
          label="BUY NOW"
         sx={{backgroundColor:
            type === 'buy'
              ? theme.palette.primary.main
              : theme.palette.primary.warning300
          }}
          onClick={() => {
            onClick(btcValue)
          }}
        />
      </TotalBox>
    </StyledBox>
  )
}

export default CheckoutCard
