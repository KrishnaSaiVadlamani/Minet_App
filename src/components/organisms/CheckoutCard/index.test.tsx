import renderer from 'react-test-renderer';
import CheckoutCard from '.';
import '@testing-library/jest-dom'
import React from 'react';
import {render,screen} from '@testing-library/react'

const handleChange = jest.fn()
it('renders buy checkout card correctly', () => {
    const tree = renderer
      .create(<CheckoutCard type='buy' btcValue={0.0234510} onClick={handleChange} symbol={''}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders sell checkout card correctly', () => {
    const tree = renderer
      .create(<CheckoutCard type='sell' btcValue={0.0234510} onClick={handleChange} symbol={''}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

describe('Organisms/CheckoutCard', () => {
  test('Renders sell card correctly', () => {
      render(<CheckoutCard type='sell' btcValue={0.0234510} onClick={handleChange} symbol={''}/>)
      //screen.getByRole('')
      expect(screen.getByRole('heading')).toHaveTextContent('0.023451');
      expect(screen.getAllByRole('img')).toHaveLength(8);
      expect(screen.getByRole('button')).toHaveTextContent('SELL NOW');
  })

  test('Renders buy card correctly', () => {
      render(<CheckoutCard type='buy' btcValue={0.0234510} onClick={handleChange} symbol={''}/>)
      //screen.getByRole('')
      expect(screen.getByRole('heading')).toHaveTextContent('0.023451');
      expect(screen.getAllByRole('img')).toHaveLength(8);
      expect(screen.getByRole('button')).toHaveTextContent('BUY NOW');
  })
})