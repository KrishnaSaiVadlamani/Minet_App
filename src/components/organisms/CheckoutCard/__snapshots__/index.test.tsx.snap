// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders buy checkout card correctly 1`] = `
<div
  className="MuiBox-root css-ufn7cb"
  data-testid="checkout-card"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-tc7jh2-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <span
        className="MuiTypography-root MuiTypography-caption1 css-1rlo75h-MuiTypography-root"
        data-testid="Hello"
        style={
          {
            "color": "#7D7D89",
          }
        }
      >
        You are buying
      </span>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        data-testid="Hello"
      >
        0.023451 
      </h6>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <span
        className="MuiTypography-root MuiTypography-caption1 css-1rlo75h-MuiTypography-root"
        data-testid="Hello"
        style={
          {
            "color": "#7D7D89",
          }
        }
      >
        1 = $3,406,069.54
      </span>
    </div>
  </div>
  <div
    className="MuiBox-root css-iwz9lp"
  >
    <div>
      <div
        className="MuiBox-root css-15budcu"
        data-testid="transaction-stepper"
      >
        <div
          className="MuiBox-root css-1oyjow8"
        >
          <div
            className="MuiBox-root css-1j5om8t"
          >
            <div>
              <img
                alt="icon"
                className="css-8fono5"
                data-testid="iconComponent"
                height="32px"
                src={
                  {
                    "process": [Function],
                  }
                }
                width="32px"
              />
            </div>
          </div>
          <div
            className="MuiBox-root css-15budcu"
          >
            <span
              className="MuiTypography-root MuiTypography-caption2 css-opfv3y-MuiTypography-root"
              data-testid="Hello"
              style={
                {
                  "color": "#7D7D89",
                }
              }
            >
              Payment method
            </span>
            <p
              className="MuiTypography-root MuiTypography-body1 css-4umqm8-MuiTypography-root"
              data-testid="Hello"
              style={
                {
                  "color": "#343446",
                }
              }
            >
              Visa credit ...8845
            </p>
          </div>
        </div>
        <div>
          <img
            alt="icon"
            className="css-1jmoofa"
            data-testid="iconComponent"
            height="40px"
            src={
              {
                "process": [Function],
              }
            }
            width="40px"
          />
        </div>
        <div
          className="MuiBox-root css-1oyjow8"
        >
          <div
            className="MuiBox-root css-1j5om8t"
          >
            <div>
              <img
                alt="icon"
                className="css-8fono5"
                data-testid="iconComponent"
                height="32px"
                src={
                  {
                    "process": [Function],
                  }
                }
                width="32px"
              />
            </div>
          </div>
          <div
            className="MuiBox-root css-15budcu"
          >
            <span
              className="MuiTypography-root MuiTypography-caption2 css-opfv3y-MuiTypography-root"
              data-testid="Hello"
              style={
                {
                  "color": "#7D7D89",
                }
              }
            >
              Delivery fees
            </span>
            <p
              className="MuiTypography-root MuiTypography-body1 css-4umqm8-MuiTypography-root"
              data-testid="Hello"
              style={
                {
                  "color": "#343446",
                }
              }
            >
              0.001 
            </p>
          </div>
        </div>
        <div>
          <img
            alt="icon"
            className="css-1jmoofa"
            data-testid="iconComponent"
            height="40px"
            src={
              {
                "process": [Function],
              }
            }
            width="40px"
          />
        </div>
        <div
          className="MuiBox-root css-1oyjow8"
        >
          <div
            className="MuiBox-root css-1j5om8t"
          >
            <div>
              <img
                alt="icon"
                className="css-8fono5"
                data-testid="iconComponent"
                height="32px"
                src={
                  {
                    "process": [Function],
                  }
                }
                width="32px"
              />
            </div>
          </div>
          <div
            className="MuiBox-root css-15budcu"
          >
            <span
              className="MuiTypography-root MuiTypography-caption2 css-opfv3y-MuiTypography-root"
              data-testid="Hello"
              style={
                {
                  "color": "#7D7D89",
                }
              }
            >
              Deposit to
            </span>
            <p
              className="MuiTypography-root MuiTypography-body1 css-4umqm8-MuiTypography-root"
              data-testid="Hello"
              style={
                {
                  "color": "#343446",
                }
              }
            >
               Wallet
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-dlxwab-MuiStack-root"
  >
    <div
      className="css-m69qwo-MuiStack-root"
    >
      <span
        className="MuiTypography-root MuiTypography-overline css-1a2m3r9-MuiTypography-root"
        data-testid="Hello"
      >
        0.023451 
      </span>
      <div>
        <img
          alt="icon"
          className="css-omu3pj"
          data-testid="iconComponent"
          height="5px"
          src={
            {
              "process": [Function],
            }
          }
          width="330px"
        />
      </div>
      <span
        className="MuiTypography-root MuiTypography-overline css-1tl9of-MuiTypography-root"
        data-testid="Hello"
      >
        $79,875.74
      </span>
    </div>
    <div
      className="css-m69qwo-MuiStack-root"
    >
      <span
        className="MuiTypography-root MuiTypography-overline css-1a2m3r9-MuiTypography-root"
        data-testid="Hello"
      >
        transaction fee
      </span>
      <div>
        <img
          alt="icon"
          className="css-a7jzyd"
          data-testid="iconComponent"
          height="5px"
          src={
            {
              "process": [Function],
            }
          }
          width="333px"
        />
      </div>
      <span
        className="MuiTypography-root MuiTypography-overline css-1tl9of-MuiTypography-root"
        data-testid="Hello"
      >
        $1,000.00
      </span>
    </div>
    <div
      className="css-m69qwo-MuiStack-root"
    >
      <span
        className="MuiTypography-root MuiTypography-overline css-1a2m3r9-MuiTypography-root"
        data-testid="Hello"
      >
        Total
      </span>
      <div>
        <img
          alt="icon"
          className="css-h8nm0t"
          data-testid="iconComponent"
          height="5px"
          src={
            {
              "process": [Function],
            }
          }
          width="350px"
        />
      </div>
      <p
        className="MuiTypography-root MuiTypography-body1 css-1xlhzrv-MuiTypography-root"
        data-testid="Hello"
      >
        $80,875.74
      </p>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1lvca97-MuiButtonBase-root-MuiButton-root"
      data-testid="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      BUY NOW
    </button>
  </div>
</div>
`;

exports[`renders sell checkout card correctly 1`] = `
<div
  className="MuiBox-root css-ufn7cb"
  data-testid="checkout-card"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-tc7jh2-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <span
        className="MuiTypography-root MuiTypography-caption1 css-1rlo75h-MuiTypography-root"
        data-testid="Hello"
        style={
          {
            "color": "#7D7D89",
          }
        }
      >
        You are selling
      </span>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        data-testid="Hello"
      >
        0.023451 
      </h6>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <span
        className="MuiTypography-root MuiTypography-caption1 css-1rlo75h-MuiTypography-root"
        data-testid="Hello"
        style={
          {
            "color": "#7D7D89",
          }
        }
      >
        1 = $3,406,069.54
      </span>
    </div>
  </div>
  <div
    className="MuiBox-root css-iwz9lp"
  >
    <div>
      <div
        className="MuiBox-root css-15budcu"
        data-testid="transaction-stepper"
      >
        <div
          className="MuiBox-root css-1oyjow8"
        >
          <div
            className="MuiBox-root css-1j5om8t"
          >
            <div>
              <img
                alt="icon"
                className="css-8fono5"
                data-testid="iconComponent"
                height="32px"
                src={
                  {
                    "process": [Function],
                  }
                }
                width="32px"
              />
            </div>
          </div>
          <div
            className="MuiBox-root css-15budcu"
          >
            <span
              className="MuiTypography-root MuiTypography-caption2 css-opfv3y-MuiTypography-root"
              data-testid="Hello"
              style={
                {
                  "color": "#7D7D89",
                }
              }
            >
              Payment method
            </span>
            <p
              className="MuiTypography-root MuiTypography-body1 css-4umqm8-MuiTypography-root"
              data-testid="Hello"
              style={
                {
                  "color": "#343446",
                }
              }
            >
              Visa credit ...8845
            </p>
          </div>
        </div>
        <div>
          <img
            alt="icon"
            className="css-1jmoofa"
            data-testid="iconComponent"
            height="40px"
            src={
              {
                "process": [Function],
              }
            }
            width="40px"
          />
        </div>
        <div
          className="MuiBox-root css-1oyjow8"
        >
          <div
            className="MuiBox-root css-1j5om8t"
          >
            <div>
              <img
                alt="icon"
                className="css-8fono5"
                data-testid="iconComponent"
                height="32px"
                src={
                  {
                    "process": [Function],
                  }
                }
                width="32px"
              />
            </div>
          </div>
          <div
            className="MuiBox-root css-15budcu"
          >
            <span
              className="MuiTypography-root MuiTypography-caption2 css-opfv3y-MuiTypography-root"
              data-testid="Hello"
              style={
                {
                  "color": "#7D7D89",
                }
              }
            >
              Delivery fees
            </span>
            <p
              className="MuiTypography-root MuiTypography-body1 css-4umqm8-MuiTypography-root"
              data-testid="Hello"
              style={
                {
                  "color": "#343446",
                }
              }
            >
              0.001 
            </p>
          </div>
        </div>
        <div>
          <img
            alt="icon"
            className="css-1jmoofa"
            data-testid="iconComponent"
            height="40px"
            src={
              {
                "process": [Function],
              }
            }
            width="40px"
          />
        </div>
        <div
          className="MuiBox-root css-1oyjow8"
        >
          <div
            className="MuiBox-root css-1j5om8t"
          >
            <div>
              <img
                alt="icon"
                className="css-8fono5"
                data-testid="iconComponent"
                height="32px"
                src={
                  {
                    "process": [Function],
                  }
                }
                width="32px"
              />
            </div>
          </div>
          <div
            className="MuiBox-root css-15budcu"
          >
            <span
              className="MuiTypography-root MuiTypography-caption2 css-opfv3y-MuiTypography-root"
              data-testid="Hello"
              style={
                {
                  "color": "#7D7D89",
                }
              }
            >
              Deposit to
            </span>
            <p
              className="MuiTypography-root MuiTypography-body1 css-4umqm8-MuiTypography-root"
              data-testid="Hello"
              style={
                {
                  "color": "#343446",
                }
              }
            >
               Wallet
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-dlxwab-MuiStack-root"
  >
    <div
      className="css-m69qwo-MuiStack-root"
    >
      <span
        className="MuiTypography-root MuiTypography-overline css-1a2m3r9-MuiTypography-root"
        data-testid="Hello"
      >
        0.023451 
      </span>
      <div>
        <img
          alt="icon"
          className="css-omu3pj"
          data-testid="iconComponent"
          height="5px"
          src={
            {
              "process": [Function],
            }
          }
          width="330px"
        />
      </div>
      <span
        className="MuiTypography-root MuiTypography-overline css-1tl9of-MuiTypography-root"
        data-testid="Hello"
      >
        $79,875.74
      </span>
    </div>
    <div
      className="css-m69qwo-MuiStack-root"
    >
      <span
        className="MuiTypography-root MuiTypography-overline css-1a2m3r9-MuiTypography-root"
        data-testid="Hello"
      >
        transaction fee
      </span>
      <div>
        <img
          alt="icon"
          className="css-a7jzyd"
          data-testid="iconComponent"
          height="5px"
          src={
            {
              "process": [Function],
            }
          }
          width="333px"
        />
      </div>
      <span
        className="MuiTypography-root MuiTypography-overline css-1tl9of-MuiTypography-root"
        data-testid="Hello"
      >
        $1,000.00
      </span>
    </div>
    <div
      className="css-m69qwo-MuiStack-root"
    >
      <span
        className="MuiTypography-root MuiTypography-overline css-1a2m3r9-MuiTypography-root"
        data-testid="Hello"
      >
        Total
      </span>
      <div>
        <img
          alt="icon"
          className="css-h8nm0t"
          data-testid="iconComponent"
          height="5px"
          src={
            {
              "process": [Function],
            }
          }
          width="350px"
        />
      </div>
      <p
        className="MuiTypography-root MuiTypography-body1 css-1xlhzrv-MuiTypography-root"
        data-testid="Hello"
      >
        $78,875.74
      </p>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1geq681-MuiButtonBase-root-MuiButton-root"
      data-testid="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      SELL NOW
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
